libs/platform/experience/src/injectables/default-icon.injectable.ts

```diff
@@ -1,15 +1,36 @@
 import { AppRef } from '@spryker-oryx/core';
 import { ssrAwaiter } from '@spryker-oryx/core/utilities';
 import { resolve } from '@spryker-oryx/di';
-import { asyncValue, IconInjectable, isPromise } from '@spryker-oryx/utilities';
+import {
+  asyncValue,
+  fontInjectable,
+  IconInjectable,
+  isPromise,
+} from '@spryker-oryx/utilities';
 import { html, TemplateResult } from 'lit';
 import { unsafeHTML } from 'lit/directives/unsafe-html.js';
-import { ResourcePlugin } from '../plugins';
+import { ResourcePlugin, ThemePlugin } from '../plugins';
 
 export class DefaultIconInjectable implements IconInjectable {
   render(type: string): TemplateResult | undefined {
-    const resourcePlugin = resolve(AppRef).findPlugin(ResourcePlugin);
-    const icon = resourcePlugin?.getIcon(type);
+    const app = resolve(AppRef);
+    const mappers = app.findPlugin(ThemePlugin)?.getIcons();
+    const source = mappers?.resource.mapping?.[type]
+      ? mappers.resource
+      : mappers?.resources?.find((resource) => resource.types.includes(type))
+          ?.resource;
+    const mapper = source?.mapping[type];
+
+    if (mapper) {
+      fontInjectable.get()?.setFont(source.id);
+
+      return html`<style>
+          .${source.id} {${source.styles}}
+        </style>
+        <span class="${source.id}">${unsafeHTML(mapper)}</span> `;
+    }
```

Comment:
Wondering if we can avoid additional DOM inside oryx-icon. Could we add the class and styles to the host element?

This will makes it cleaner, but also easier to sort out the layout and additional styles.